### DISCUSSION TITLE
support `--show-config` options

### DISCUSSION THREAD

**Discussion by unional** [biomejs/biome#2712] [Ideas]
either print out the while consolidated config (after merging configs from `extends`), or able to show config for a specific target (e.g. `biome check packages/x/package.json --show-config` )

**Comment by ematipico** [comment]
Sorry, I honestly didn't understand the idea.

**Reply by unional** [reply]
It's similar to `tsc --showConfig`, which print out the configuration.

Here it prints out the biome configuration, so that you can see what configuration is in effect.
It's useful to tell why the file is not format/lint the way you expect it to.

**Reply by ematipico** [reply]
I didn't know the command. 

How is the output supposed to be? What do you propose?  What should we expect to see? Why isn't the command `rage` sufficient?

**Reply by unional** [reply]
`biome rage` is about the environment. `show-config` is about the config.

`tsc --showConfig` prints out the config in JSON.

for `biome` it would like:

```sh
> biome show-config
{
  "linter": ...
}

> biome show-config ./path/to/some/specific-file
# effective config for that path
{
  "linter" ...
}
```

**Reply by unional** [reply]
ar, I got what you mean. Didn't know `biome rage` has the `--formatter` and `--linter` options.

It would be great if `rage` supports `biome range --formatter|--linter ./path/to/specific/file`

**Reply by ematipico** [reply]
Can't you use `--config-path`?

**Reply by unional** [reply]
um... `--config-path` is used to specify which config file to use.
This is for knowing what is the config being applied when the config file is used (`biome.json` or custom with `--config-path`), after the config is resolved based on the `extends`, `overrides`, `file.include` and `file.ignore`.

**Comment by magnusriga** [comment]
This would be useful in debugging. Currently, the formatter is not working as expected given the config file in the project folder. I would like to check if biome actually found the correct config file or not.

**Reply by ematipico** [reply]
In v2 we enhanced the `rage` command, it will print the config path too

**Reply by magnusriga** [reply]
@ematipico Thank you for the quick reply. I tried it, but encountered an issue. Is the below intended or a bug (i.e. it does not use the provided config)?

> :!pnpm biome rage --config-path=/home/nfu/nfront/packages/config-biome/biome.jsonc
> CLI:
>   Version:                      2.0.0-beta.2
>   Color support:                true
> 
> Platform:
>   CPU Architecture:             aarch64
>   OS:                           linux
> 
> Environment:
>   BIOME_LOG_PATH:               unset
>   BIOME_LOG_PREFIX_NAME:        unset
>   BIOME_CONFIG_PATH:            unset
>   NO_COLOR:                     unset
>   TERM:                         xterm-ghostty
>   JS_RUNTIME_VERSION:           v22.14.0
>   JS_RUNTIME_NAME:              node
>   NODE_PACKAGE_MANAGER:         pnpm/10.10.0
> 
> Biome Configuration:
>   Status:                       Loaded successfully
>   Path:                         /home/nfu/nfront/biome.json
>   Formatter enabled:            true
>   Linter enabled:               true
>   Assist enabled:               true
>   VCS enabled:                  true
> 
> Workspace:
>   Open Documents:               0

**Reply by magnusriga** [reply]
I also tried with the env:

> :!export BIOME_CONFIG_PATH=/home/nfu/nfront/packages/config-biome/biome.jsonc; pnpm biome rage --config-path=/home/nfu/nfront/packages/config-biome/biome.jsonc
> CLI:
>   Version:                      2.0.0-beta.2
>   Color support:                true
> 
> Platform:
>   CPU Architecture:             aarch64
>   OS:                           linux
> 
> Environment:
>   BIOME_LOG_PATH:               unset
>   BIOME_LOG_PREFIX_NAME:        unset
>   BIOME_CONFIG_PATH:            /home/nfu/nfront/packages/config-biome/biome.jsonc
>   NO_COLOR:                     unset
>   TERM:                         xterm-ghostty
>   JS_RUNTIME_VERSION:           v22.14.0
>   JS_RUNTIME_NAME:              node
>   NODE_PACKAGE_MANAGER:         pnpm/10.10.0
> 
> Biome Configuration:
>   Status:                       Loaded successfully
>   Path:                         /home/nfu/nfront/biome.json
>   Formatter enabled:            true
>   Linter enabled:               true
>   Assist enabled:               true
>   VCS enabled:                  true
> 
> Workspace:
>   Open Documents:               0

**Reply by ematipico** [reply]
That's probably a bug

**Reply by magnusriga** [reply]
@ematipico Got it, thanks for confirming. Another one that looks like a bug, preventing conform.nvim from doing full formatting, lint issue fixing, and import order fixing:

https://github.com/biomejs/biome/issues/5958

**Comment by Jacse** [comment]
This functionality is especially useful in monorepo setups. I currently have a nested biome.json file but it doesn't apply the rules like I expect them to so it would be *very* useful to see what the final config looks like for that file

**Comment by mkvlrn** [comment]
I think pointing out the direct equivalent in eslint would help contextualize the idea: https://eslint.org/docs/latest/use/command-line-interface#--print-config

**Comment by nevir** [comment]
This would be extremely helpful for diagnosing complex configurations (e.g. that utilize overrides) to see what the final config is for a given file

**Comment by elektronik2k5** [comment]
`tsc`, which also supports complex configurations has this exact flag: `--showConfig  Print the final configuration instead of building.`

It is essential in complex setups.
